### ISSUE TITLE
uncertain length of tokens

### ISSUE THREAD

**Issue by orchardchang** [pytorch/fairseq#939]
When I use the following code, uncertain length of tokens makes me confusing.
```
import torch
roberta = torch.hub.load('pytorch/fairseq', 'roberta.large')
roberta.eval()

question = 'is there a baby giraffe'
tokens = roberta.encode(question)

tensor([    0,   354,    89,    10,  1928, 25650, 17459,     2])
```
The str question `'is there a baby giraffe'` have 5 words, but tokens have 6 tokens, except `<s>` and `</s>`.

I guess word `giraffe` is an unknown word, so the model uses more than one tokens to encode.

Encoding word with more than one tokens seems not appropriate. How could I avoid the problem. Thanks.

**Comment by myleott** [comment]
It’s not desirable to use a vocabulary with a 1:1 mapping to every English word, since it would be much too large and many of the words would not appear frequently enough in the training set to learn good embeddings for them.

The vocabulary inside RoBERTa instead uses Byte-Pair Encoding and has a vocabulary of 50K sub-word units. Some of these map to whole words (for common words) and in other cases it’s word pieces that get stitched together to encode a word.

What is your use case that you need a 1:1 mapping to words?

**Comment by orchardchang** [comment]
In previous Visual Question Answering model, sentences need to embed into fixed length tokens with mask for convenience. Therefore,  not 1:1 mapping may introduce some unexpected errors. For instance, when the length of tokens is longer than fixed length, I have to cut them off.

I have checked the unexpected sentences to avoid error. Thanks a lot for explaining.
